src/common/database.c

```diff
@@ -1371,6 +1371,30 @@ static gboolean pid_is_alive(int pid)
   }
 #else
   pid_is_alive = !((kill(pid, 0) == -1) && errno == ESRCH);
+
+#ifdef __linux__
+  // If this is Linux, we can query /proc to see if the pid is
+  // actually a darktable instance.
+  if (pid_is_alive)
+  {
+    char filename[64];
+    char cmdline[MAX_PATH+1];
+    sprintf(filename, "/proc/%d/cmdline", pid);
+    FILE* f = fopen(filename, "r");
+    if (f) {
+      size_t size;
+      size = fread(cmdline, sizeof(char), MAX_PATH, f);
+      if (size > 0)
+      {
+        if (strstr(cmdline, "darktable") == NULL) {
```

Comment:
The problem I see is having some other script inside the darktable folder which is running with that PID. It would trigger the test ("darktable" is in the directory name). However, that is such a fringe corner case that we can ignore it.

If we wanted to do the right thing we would have to check /proc/<pid>/fd/* and see if any one of those is pointing to the sqlite file. That's just a remark, please don't add code like that. :-)